### PR TITLE
SONARKT-235 Improve spotless configuration

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**
* Update spotless to the latest version 6.11.0
* configure spotless for all modules except kotlin-checks-test-sources

### CODE REVIEW

**Comment by margarita-nedzelska-sonarsource** [comment]
I haven't changed what was configured before. Just applied it to the whole project.

File: build.gradle.kts [hunk 1]
```diff
@@ -9,13 +9,53 @@ plugins {
...
+        lineEndings = com.diffplug.spotless.LineEnding.UNIX
+
+        fun SourceSet.findSourceFilesToTarget() = allJava.srcDirs.flatMap { srcDir ->
+            project.fileTree(srcDir).filter { file ->
+                file.name.endsWith(".kt") || (file.name.endsWith(".java") && file.name != "package-info.java")
+            }
+        }
+
+        kotlin {
+            // ktlint()
```
Comment (dorian-burihabwa-sonarsource) [inline build.gradle.kts:34]: Can we remove this or add a comment explaining why there might have been a good reason to call `ktlint`?

Comment (dorian-burihabwa-sonarsource) [reply build.gradle.kts:34]:
```suggestion
```

File: sonar-kotlin-plugin/build.gradle.kts [hunk 2]
```diff
@@ -80,8 +44,12 @@ test.dependsOn(project(":kotlin-checks-test-sources").tasks.named("build"))
...
 tasks.jar {
     manifest {
-        val displayVersion = if (project.property("buildNumber") == null) project.version else project.version.toString()
-            .substring(0, project.version.toString().lastIndexOf(".")) + " (build ${project.property("buildNumber")})"
+        val displayVersion = if (project.property("buildNumber") == null) {
+            project.version
+        } else {
+            project.version.toString()
+                .substring(0, project.version.toString().lastIndexOf(".")) + " (build ${project.property("buildNumber")})"
+        }
```
Comment (dorian-burihabwa-sonarsource) [inline sonar-kotlin-plugin/build.gradle.kts:52]: Thanks, that's a lot easier to read 👍🏿

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM! If possible, it would be nice to remove the commented ktlint line or add some more information about why we should keep it here. Do we need to run it for every release? every year? whenever we add a new module?

File: build.gradle.kts (hunk 1 above)
Comment (margarita-nedzelska-sonarsource) [reply build.gradle.kts:34]: I think we wanted to use ktlint at some point but then decided not to, as we were not happy with the result.

I think it's okay to drop the line and re-launch the discussion in the future if we want to use it or not.

**Review by johann-beleites-sonarsource — COMMENTED** [review]
Nice!

**Pull request merged**